brokers/unified/vscode/broker.go

```diff
@@ -245,8 +245,9 @@ func extensionOrURL(extensionOrURL string) (extension string, URL string) {
 	}
 }
 
-func (b *brokerImpl) generatePluginArchiveName(plugin model.ChePlugin) string {
-	return fmt.Sprintf("%s.%s.%s.%s", plugin.Publisher, plugin.Name, plugin.Version, b.rand.String(10))
+func (b *brokerImpl) generatePluginArchiveName(plugin model.ChePlugin, archivePath string) string {
```

Comment:
This likely won't work on current master; `archivePath` is generated and no longer resolves filename, and resolving filename from URL won't work for many URLs.

Reply:
Ah nevermind I see this commit also modifies how `archivePath` is generated.

I worry that depending on URLs ending in the filename will cause problems -- e.g. https://marketplace.visualstudio.com/_apis/public/gallery/publishers/vscjava/vsextensions/vscode-java-debug/0.16.0/vspackage is a valid URL.

Reply:
This is fixed with commit https://github.com/eclipse/che-plugin-broker/pull/60/commits/ce4c699d7b92f7fdbbac33bfaee10e59532bcc8b